### DISCUSSION TITLE
Proposal to add 'archetype' field to path specifications

### DISCUSSION THREAD

**Discussion by myronww** [OAI/OpenAPI-Specification#3371] [Enhancements]
I would like to propose the addition of an 'archetype' field to path specifications to allow for the declaration of the architectural pattern that a path implements.  There are two major reasons for this that I lay out in my proposal.

1. It is vital information for automated testing tools to allow tools to scan APIs and ensure that they meet architectural standards.  This allows architectural or design constraints to be tested and applied across a larger API set.
2. It is very useful information for code generators and or dynamic clients that want to create interop code or other means of automatically building communication mechanisms with an api.

https://github.com/OAI/OpenAPI-Specification/pull/3369

**Comment by handrews** [comment]
This is being more actively discussed in moonwalk, so I'm going to close this discussion in favor of:

* https://github.com/OAI/sig-moonwalk/discussions/79
